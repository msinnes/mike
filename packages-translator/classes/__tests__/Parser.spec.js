const iParser = require('@mike/translator-interfaces/iParser');

const Parser = require('../Parser');

describe('Parser', () => {
  it('should be an abstract class', () => {
    expect(() => {
      Parser('string');
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new Parser('string');
    }).toThrowErrorMatchingSnapshot();
  });

  it('should implement iParser', () => {
    expect(Parser.implements(iParser)).toBe(true);
  });
});