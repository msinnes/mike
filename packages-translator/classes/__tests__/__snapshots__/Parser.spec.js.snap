// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser instance instance.parse should throw an error if no test is passed in 1`] = `"RuntimeError: parsers can only parse string input"`;

exports[`Parser instance instance.wrapRule should throw a runtime error if the rule does not return an AstNode 1`] = `"RuntimeError: Syntax Rules should return an instance of AstNode"`;

exports[`Parser should be an abstract class 1`] = `"ClassError: Cannot call a class as a function"`;

exports[`Parser should be an abstract class 2`] = `"ClassError: Abstract classes cannot be instantiated"`;
